### ISSUE TITLE
UCSC Issues

### ISSUE THREAD

**Issue by denis-yuen** [CancerCollaboratory/sandbox#140]
A number of TODO items in the main README that probably make more sense as issues
```
## TODO

Consonance is a work in progress, there are many features and bugs that need to be fleshed out. Here are the most imporant issues.

* can the indivdual images be hosted on Quay.io so the install_boostrap doesn't need to spend time building them?
* there are a ton of settings that the bootstrapper should expose to end users.  Regions, AMI, etc
* need to setup the previous job hash and expose that through as a callable parameter to the consonance command line
* should generate new token for admin on each deployment, should also use better passwords assigned at build time for rabbitmq and postgres
* it seems like the instance type is hard-coded for Youxia yet it's a param for Consonance.  It really should be a param otherwise a given deployment will only work for a particular AMI/instance type.
* re-enqueing work for lost jobs ultimately fails
* we need to improve the worker deamon, it fails to launch properly on occasion which menas we have workers that never process jobs
* we need to monitor worker instances/daemons. If the worker daemon on a given worker instance has not sent a ping in a while we need to reap the instance.  This will prevent the situation where a woker daemon has died on a worker host but it's kept around since there are running jobs.
* a large number of submissions will eventually see the provisioner stop provisioning new hosts, even when the total running is under the total fleet size. I saw this when all workers running where zombie workers e.g. the worker daemons had crashed so nothing was scheduled to these nodes.  Once I restarted the worker daemons, the provisioner started provisioning new additional nodes.  Very strange.
* seems like the provisioner will not detect that a VM has been terminated... doesn't detect `State: {Code: 48,Name: terminated}`
* Walt items:
    * saw single job not being picked up, job present in queue, DB says START
    * worker daemon crashes and VM continues running
    * stderr and stdout are merged in consonance status call
    * need to be able to cancel a job and or resubmit a job
    * consonance only starts a VM flavor that is in consonance config; whether it is in correct Rabbit queue or not
```

**Comment by denis-yuen** [comment]
re: hosting launcher images on quay.io 

The current images cannot be uploaded since they would contain keys and credentials. Sanitized versions probably could and would speed up the build. Start time is only about a minute though, including manual input. 

re: Regions and AMI settings

These seem to already be exposed, possibly added after these notes.

**Comment by denis-yuen** [comment]
re: it seems like the instance type is hard-coded for Youxia yet it's a param for Consonance.  It really should be a param otherwise a given deployment will only work for a particular AMI/instance type.

@briandoconnor  FYI, was looking at this issue. It looks like total nodes was hard-coded in this commit
https://github.com/Consonance/consonance/pull/133/commits/67d65533f3475b2f93070a73d2c16c0586b756c5 but this might have been a misunderstanding since this forces youxia to deploy a "total number of nodes, all of one instance type".  

In other words, youxia can deploy one instance type up to total-num-nodes or it can deploy a mixed set of instance types specified via a map. The former overrides the later which is why it seemed like instance type was hard-coded

If you see this issue again, this should be fixed in the "else" code dealing with instanceRestrictions or at a higher level by modifying batch-size (which does control the number of nodes deployed at a time).
